### PR TITLE
Clean DB properties on close

### PR DESCRIPTION
**Pull request by agroszer**
otherwise persistent object seem to hang around
noticed by our tests

### CODE REVIEW

File: src/ZODB/DB.py [hunk 1]
```diff
@@ -645,6 +645,15 @@ def _(c):
         self._mvcc_storage.close()
         del self.storage
         del self._mvcc_storage
+        # clean up references to other DBs
+        self.databases.clear()
+        # clean up the connection pool
+        while self.pool.pop():
+            pass
```
Comment (tseaver) [inline src/ZODB/DB.py]: Maybe `AbstractConnectionPool` should grow a `clear()` method:  we would then just call it for each one:

```python
        self.pool.clear()
        self.historical_pool.clear()
```

File: src/ZODB/tests/multidb.txt [hunk 2]
```diff
@@ -137,7 +137,7 @@ an exception:
 
 Clean up:
 
-    >>> for a_db in dbmap.values():
+    >>> for a_db in list(dbmap.values()):
```
Comment (tseaver) [inline src/ZODB/tests/multidb.txt]: Nice find.

File: src/ZODB/DB.py [hunk 3]
```diff
@@ -645,6 +645,15 @@ def _(c):
         self._mvcc_storage.close()
         del self.storage
         del self._mvcc_storage
+        # clean up references to other DBs
+        self.databases.clear()
```
Comment (jamadden) [inline src/ZODB/DB.py]: I wonder if this should be `self.databases = {}` or even `self.databases = None`? That way if someone else has a reference to the databases dict (not hard if you're manually constructing a multi-db) it doesn't get cleared out from under them.

Comment (jimfulton) [inline src/ZODB/DB.py]: +1

Comment (jimfulton) [inline src/ZODB/DB.py]: In fact, I just looked at some code that clearing the dict would break.

@agroszer just reset the attr as @jamadden suggested.

Comment (strichter) [inline src/ZODB/DB.py]: Done.

File: src/ZODB/DB.py (hunk 1 above)
Comment (strichter) [inline src/ZODB/DB.py]: Done. It was easy enough.

**Comment by strichter** [comment]
@jamadden & @tseaver Are you okay with my implementation of your comments? I would love to see this getting merged and a release being made. :smile:

**Comment by jimfulton** [comment]
Gaaaa, we didn't check for an entry in CHANGES.rst. I just added one.

**Comment by jamadden** [comment]
Snap. Sorry I missed that.

**Comment by jimfulton** [comment]
Released in 5.2.4.

**Pull request merged**